### PR TITLE
[ruby 2.4] Fixes for Integer#digits in PR #4375

### PR DESCRIPTION
**Pull request by whwilder**
I've split the implementation for Integer#digits amongst Fixnum and Bignum, which should fix the issues in #4375 

ref #4293

### CODE REVIEW

File: core/src/main/java/org/jruby/RubyFixnum.java [hunk 1]
```diff
@@ -302,6 +302,53 @@ public IRubyObject times(ThreadContext context, Block block) {
         return RubyEnumerator.enumeratorizeWithSize(context, this, "times", timesSizeFn(context.runtime));
     }
 
+    /** rb_fix_digits
+     *
+     */
+    @Override
+    public RubyArray digits(ThreadContext context, IRubyObject base) {
+
+        long self = ((RubyFixnum)this).getLongValue();
```
Comment (headius) [inline core/src/main/java/org/jruby/RubyFixnum.java]: This is not necessary...`this` is already known to be RubyFixnum.

File: core/src/main/java/org/jruby/RubyBignum.java [hunk 2]
```diff
@@ -236,6 +236,54 @@ public static BigInteger long2big(long arg) {
      *  ================
      */
 
+    /** rb_big_digits
+     *
+     */
+    @Override
+    public RubyArray digits(ThreadContext context, IRubyObject base) {
+        BigInteger self = (this).getValue();
```
Comment (headius) [inline core/src/main/java/org/jruby/RubyBignum.java]: You could just access the field here, or at least call it without the parens. I assume this is just a mistake converting the RubyFixnum-only logic.

File: core/src/main/java/org/jruby/RubyBignum.java [hunk 3]
```diff
@@ -236,6 +236,54 @@ public static BigInteger long2big(long arg) {
...
+        if (self.compareTo(new BigInteger("0")) == -1) {
+            throw context.getRuntime().newMathDomainError("out of domain");
```
Comment (headius) [inline core/src/main/java/org/jruby/RubyBignum.java]: Throw context.runtime into a local variable at the top and reuse it, in the style of other methods.

**Comment by whwilder** [comment]
Thanks for the help!  Squashed a few other style issues too.

**Comment by whwilder** [comment]
@headius Is there anything preventing this PR from being merged, or is it just a low priority for the moment?

**Comment by kares** [comment]
@whwilder great work - wanted to get `#digits` in as well, thanks :)

**Pull request merged**